source/common/http/http1/codec_impl.cc

```diff
@@ -1115,6 +1139,27 @@ int ClientConnectionImpl::onHeadersComplete() {
       }
     }
 
+    // https://tools.ietf.org/html/rfc7230#section-3.3.3
+    // If a message is received with both a Transfer-Encoding and a
+    // Content-Length header field, the Transfer-Encoding overrides the
+    // Content-Length. Such a message might indicate an attempt to
+    // perform request smuggling (Section 9.5) or response splitting
+    // (Section 9.4) and ought to be handled as an error. A sender MUST
+    // remove the received Content-Length field prior to forwarding such
+    // a message.
+
+    // Reject response if Transfer-Encoding and Content-Length headers are present
+    // or remove Content-Length and serve response if allowed by http1 codec settings.
+    if (parser_.uses_transfer_encoding != 0 && headers->ContentLength()) {
+      if ((parser_.flags & F_CHUNKED) && codec_settings_.allow_chunked_length_) {
+        headers->removeContentLength();
+      } else {
+        sendProtocolError(Http1ResponseCodeDetails::get().ContentLengthNotAllowed);
```

Comment:
I don't think sending protocol error makes sense as a client. All we can do is raise a protocol exception and stat it.

Reply:
@mattklein123 I'm a bit confused as other checks above and below this code call sendProtocolError and throw CodecProtocolException